js/src/ui/Form/AddressSelect/addressSelect.js

```diff
@@ -15,262 +15,799 @@
   }
 
-  renderItem = (entry) => {
-    const { address, name } = entry;
+  renderCurrentInput () {
+    if (!this.props.allowInput) {
+      return null;
+    }
 
-    const _balance = this.getBalance(address);
-    const balance = _balance ? _balance.toNumber() : _balance;
+    const { inputValue } = this.state;
 
-    if (!this.items[address] || this.items[address].balance !== balance) {
-      this.items[address] = {
-        text: name && name.toUpperCase() || address,
-        value: this.renderMenuItem(address),
-        address, balance
-      };
+    if (inputValue.length === 0 || !/^(0x)?[a-f0-9]*$/i.test(inputValue)) {
+      return null;
     }
 
-    return this.items[address];
+    return (
+      <div
+
+      >
+        { this.renderAccountCard({ address: inputValue }) }
+      </div>
+    );
   }
 
-  getBalance (address) {
-    const { balances = {} } = this.props;
-    const balance = balances[address];
+  renderCloseIcon () {
+    const { expanded } = this.state;
 
-    if (!balance) {
+    if (!expanded) {
       return null;
     }
 
-    const ethToken = balance.tokens.find((tok) => tok.token && tok.token.tag && tok.token.tag.toLowerCase() === 'eth');
+    return (
+      <div className={ styles.closeIcon } onClick={ this.handleClose }>
+        <CloseIcon style={ { width: 48, height: 48 } } />
+      </div>
+    );
+  }
 
-    if (!ethToken) {
-      return null;
+  renderAccounts () {
+    const { values } = this.state;
+
+    if (values.length === 0) {
+      return (
+        <div className={ styles.categories }>
+          <div className={ styles.empty }>
+            No account matches this query...
+          </div>
+        </div>
+      );
     }
 
-    return ethToken.value;
+    const categories = values.map((category) => {
+      return this.renderCategory(category.label, category.values);
+    });
+
+    return (
+      <div className={ styles.categories }>
+        { categories }
+      </div>
+    );
   }
 
-  renderBalance (address) {
-    const balance = this.getBalance(address);
-    const value = fromWei(balance);
+  renderCategory (name, values = []) {
+    if (values.length === 0) {
+      return null;
+    }
+
+    const cards = values
+      .map((account) => this.renderAccountCard(account));
 
     return (
-      <div className={ styles.balance }>
-        { value.toFormat(3) }<small> { 'ETH' }</small>
+      <div className={ styles.category } key={ name }>
+        <div className={ styles.title }>{ name }</div>
+        <div className={ styles.cards }>
+          <div>{ cards }</div>
+        </div>
       </div>
     );
   }
 
-  renderMenuItem (address) {
-    const balance = this.props.balances
-      ? this.renderBalance(address)
+  renderAccountCard (_account) {
+    const { copied } = this.state;
+    const { address, index = null } = _account;
+
+    const account = this.props.accountsInfo[address];
+    const name = (account && account.name && account.name.toUpperCase()) || address;
```

Comment:
This is just to get the name actually, which isn't displayed as in other parts of the UI (which is the goal of the PR actually : having a new component/way of presenting things)